### PR TITLE
Remove escape-goat package

### LINKED ISSUE #7175: vendor bundle error in IE11
**Describe the bug**

There appears to be an error in the Simorgh vendor bundle as shown in the screenshot

Screenshot 2020-07-08 at 10 41 31

If you click on the link to the error it takes you to a package called escape goat

Screenshot 2020-07-08 at 10 41 40

I think this error occurs because IE11 does not support arrow functions meaning that this package is not transpiled to es5

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to BrowserStack and load https://www.test.bbc.com/zhongwen/simp in IE11
2. Open devtools console and observe error

**Expected behaviour**
No errors in the console in IE11

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by chris-hinds**
Resolves #7175 

**Overall change:**
- Removes the escape-goat package
- Copies the required function from escape goat and create a new util function around it
- Adds unit test for new util function

Using escape-goat package in IE11

image

After using single function from escape-goat package (Note the errors here are from the Ads module loading on the development build)

image

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added labels to this PR for the relevant pod(s) affected by these changes
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by joshcoventry — APPROVED** [review]

File: src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/getImageBlock.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ const getImageBlock = (type, blockData, isAmp) => {
   const { altText, dimensions } = imageData;
 
   return {
-    alt: htmlUnescape(altText),
+    alt: escape(altText),
```
Comment (andrewscfc) [inline src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/getImageBlock.js]: Do we even need to do this anymore, I remember @jamesdonoh making these changes that removed the need for escaping in block data (I think?) https://github.com/bbc/simorgh/commit/5d31ea66e902da5d5c348656d6e4c86b24a32b96 

I may be wrong but can we check we even need to escape?

Comment (chris-hinds) [reply src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/getImageBlock.js]: we did as of a week ago, this was a bug picked up in the alt text captions I think.

**Review by oluoluoxenfree — APPROVED** [review]

**Comment by chris-hinds** [comment]
This seems to have completely broken story pages........

**Review by rhenshaw56 — COMMENTED** [review]
page is broken for me while running a live build

Screen Shot 2020-07-08 at 16 46 32

File: src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/getImageBlock.js (hunk 1 above)
Comment (andrewscfc) [reply src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/getImageBlock.js]: Hmm, I must be missing something but seems a safe change

**Review by andrewscfc — APPROVED** [review]

**Comment by chris-hinds** [comment]
> page is broken for me while running a live build
> 
> Screen Shot 2020-07-08 at 16 46 32

Good spot @rhenshaw56 I didn't test this on a story page where this code is actually run, I have fixed the import and the page is loading now.

**Review by joshcoventry — APPROVED** [review]

File: src/app/routes/utils/htmlUnescape/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Function taken from https://github.com/sindresorhus/escape-goat as this package is not compiled down to ES5 so does not support IE11
```
Comment (simonsinclair) [inline src/app/routes/utils/htmlUnescape/index.js:1]: 👍

**Review by simonsinclair — APPROVED** [review]

**Review by joshcoventry — APPROVED** [review]

File: src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/getImageBlock.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { htmlUnescape } from 'escape-goat';
+import htmlUnescape from '../../../../../utils/htmlUnescape';
```
Comment (andrewscfc) [inline src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/getImageBlock.js:1]: Is there no alias for `utils`?

**Review by andrewscfc — APPROVED** [review]

**Review by rhenshaw56 — APPROVED** [review]
Tested as per the description in this issue https://github.com/bbc/simorgh/issues/6998 and this is working fine again.

**Comment by paruchurisilpa** [comment]
Looks good to me, don't see any console errors in IE11 and other browsers..

**Pull request merged**